tests/unit/test_tiler.py

```diff
@@ -455,60 +536,95 @@ def it_can_calculate_n_tiles_row(
-                False,
-                [True, True],
-                2,
+                [False, False],
+                0,
             ),
             (
                 Tile(PILIMG.RGBA_COLOR_500X500_155_249_240, CP(0, 10, 0, 10)),
                 Tile(PILIMG.RGBA_COLOR_500X500_155_249_240, CP(0, 10, 0, 10)),
-                False,
-                [False, False],
-                2,
+                [True, False],
+                1,
             ),
+        ),
+    )
+    def it_can_generate_grid_tiles_with_check_tissue(
+        self,
+        request,
+        tmpdir,
+        tile1,
+        tile2,
+        has_enough_tissue,
+        expected_n_tiles,
+    ):
+        tmp_path_ = tmpdir.mkdir("myslide")
+        image = PILIMG.RGBA_COLOR_500X500_155_249_240
+        image.save(os.path.join(tmp_path_, "mywsi.png"), "PNG")
+        slide_path = os.path.join(tmp_path_, "mywsi.png")
+        slide = Slide(slide_path, "processed")
+        _extract_tile = method_mock(request, Slide, "extract_tile")
+        _has_enough_tissue = method_mock(request, Tile, "has_enough_tissue")
+        _has_enough_tissue.side_effect = has_enough_tissue
+        _grid_coordinates_generator = method_mock(
+            request, GridTiler, "_grid_coordinates_generator"
+        )
+        _grid_coordinates_generator.return_value = [CP(0, 10, 0, 10), CP(0, 10, 0, 10)]
+        _extract_tile.side_effect = [tile1, tile2]
+        grid_tiler = GridTiler((10, 10), level=0, check_tissue=True, tissue_percent=60)
+        tiles = [tile1, tile2]
+
+        generated_tiles = list(grid_tiler._tiles_generator(slide))
+
+        _grid_coordinates_generator.assert_called_once_with(grid_tiler, slide)
+        assert _extract_tile.call_args_list == (
+            [call(slide, CP(0, 10, 0, 10), 0), call(slide, CP(0, 10, 0, 10), 0)]
+        )
+        assert _has_enough_tissue.call_args_list == [call(tile1, 60), call(tile2, 60)]
+        assert len(generated_tiles) == expected_n_tiles
+        for i, tile in enumerate(generated_tiles):
+            assert tile[0] == tiles[i]
+
+    @pytest.mark.parametrize(
+        "tile1, tile2, has_enough_tissue, expected_n_tiles",
+        (
             (
                 Tile(PILIMG.RGBA_COLOR_500X500_155_249_240, CP(0, 10, 0, 10)),
                 Tile(PILIMG.RGBA_COLOR_500X500_155_249_240, CP(0, 10, 0, 10)),
-                True,
-                [False, False],
-                0,
+                [True, True],
+                2,
             ),
             (
                 Tile(PILIMG.RGBA_COLOR_500X500_155_249_240, CP(0, 10, 0, 10)),
                 Tile(PILIMG.RGBA_COLOR_500X500_155_249_240, CP(0, 10, 0, 10)),
-                True,
-                [True, False],
-                1,
+                [False, False],
+                2,
             ),
         ),
     )
-    def it_can_generate_grid_tiles(
+    def it_can_generate_random_tiles_with_no_check_tissue(
```

Comment:
i think the name is incorrect, why the test is asking for a grid tiler while the name mentions the random one? Typo?